.golangci.yml

```diff
@@ -12,371 +12,135 @@
+    - perfsprint
+    - prealloc
+    - predeclared
+    - promlinter
+    - protogetter
+    - reassign
+    - revive
+    - rowserrcheck
+    - sloglint
+    - spancheck
+    - sqlclosecheck
+    - staticcheck
+    - stylecheck
+    - tagalign
+    - tagliatelle
+    - tenv
+    - testableexamples
+    - testifylint
+    - thelper
+    - tparallel
+    - unconvert
+    - unparam
     - unused
-  fast: false
-
-severity:
-  # Default value is empty string.
-  # Set the default severity for issues. If severity rules are defined and the issues 
-  # do not match or no severity is provided to the rule this will be the default 
-  # severity applied. Severities should match the supported severity names of the 
-  # selected out format.
-  # - Code climate: https://docs.codeclimate.com/docs/issues#issue-severity
-  # -   Checkstyle: https://checkstyle.sourceforge.io/property_types.html#severity
-  # -       Github: https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message
-  default-severity: error
-
-  # The default value is false. 
-  # If set to true severity-rules regular expressions become case sensitive.
-  case-sensitive: false
-
-  # Default value is empty list.
-  # When a list of severity rules are provided, severity information will be added to lint
-  # issues. Severity rules have the same filtering capability as exclude rules except you 
-  # are allowed to specify one matcher per severity rule.
-  # Only affects out formats that support setting severity information.
-  rules:
-    - linters:
-      - dupl
-      severity: info
+    - usestdlibvars
+    - wastedassign
+    - whitespace
+    - zerologlint
+  disable:
+    - govet # We already run with `make verify/govet`
+    # We do not use
+    - cyclop # Cyclomatic complexity
+    - depguard # We don't guard against dependencies
+    - dupl # Tracks code duplication. Too much duplication in tests. False positives in non-tests
+    - exhaustruct # Explicitly instantiating all structs is painful for K8s structs
+    - funlen # Long func names happen
+    - gocognit # Cognitive complexity
+    - gofumpt # We don't rely on gofumpt
+    - gomnd # Magic Number Detection. Many false positives.
+    - gomoddirectives # We need `replace` in `go.mod`
+    - interfacebloat # No more than 10 interface methods
+    - ireturn # Accept interfaces return concrete types
+    - lll # Limit line length
+    - maintidx # Maintainability index
+    - mnd # Magic Number Detection. Many false positives
+    - nestif # Don't allow too many nested if statements
+    - nlreturn # Always have empty line before return
+    - nonamedreturns
+    - testpackage # Require separate test package to catch leaky unexported dependencies
+    - varnamelen # Long var names happen
+    - wsl # Too strict of a whitespace linter
+    # Consider adding in future
+    - err113 # Do not create errors dynamically from scratch. Instead, wrap static (package-level) error.
```

Comment:
This linter is a 2-3 point task on its own to add meaningful error wraps for external package errors.